src/ds5/ds5-options.h

```diff
@@ -376,8 +377,68 @@ namespace librealsense
         lazy<option_range> _range;
         hw_monitor& _hwm;
         sensor_base* _sensor;
+        limits_option* _gain_limit_enable;
     };
 
+    // Auto-Limits Enable/ Disable
+    class limits_option : public option
+    {
+    public:
+
+        limits_option(rs2_option option, option_range range, const char* description, hw_monitor& hwm) :
+            _option(option), _toggle_range(range), _description(description), _hwm(hwm) 
+        {
+            _value = 0;
+        };
```

Comment:
This control should store the handle to the `auto_gain_limit_option` , so when toggled on, it will use the cached value and send it to FW

Reply:
I implemented it in this way  because when I sent the handler to toggle it required sending the cached value to set(..) it means changing the API.

Reply:
I implemented it in this way because when I sent the handler to toggle and changed the control limit value, it wasn't communicated to the toggle so it didn't change state (from off to on)